### PR TITLE
DataFrame 1.7.0

### PR DESCRIPTION
**Pull request by yssource**
Specify library name and version:  **DataFrame/1.7.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/dataframe/all/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 2.8.12)
+project(cmake_wrapper)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+if (MSVC AND WIN32 AND BUILD_SHARED_LIBS)
```
Comment (SSE4) [inline recipes/dataframe/all/CMakeLists.txt]: is it needed?
I see the code below:
```
if compiler == "Visual Studio" and self.options.shared:
            raise ConanInvalidConfiguration(
                "Could not support this specific configuration. Try static."
            )
```
so, either this line is useless, or code in conanfile?

File: recipes/dataframe/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+            )
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        if (
+            self.settings.os == "Windows"
+            and self.settings.compiler == "Visual Studio"
+            and self.options.shared
+        ):
+            cmake.definitions["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True
```
Comment (SSE4) [inline recipes/dataframe/all/conanfile.py]: again, this seems to be unreachable (also, duplicated in CMakeLists.txt)

File: recipes/dataframe/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("License", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        if self.options.shared:
```
Comment (SSE4) [inline recipes/dataframe/all/conanfile.py]: are manual copies needed here? as you call `cmake.install` above

**Comment by yssource** [comment]
@SSE4  
I thought it was supposed to support with the dynamic libraries for MSVC, but it fails although I try to do a patch for it.
However, actually the upstream does not support it.

File: recipes/dataframe/all/CMakeLists.txt (hunk 1 above)
Comment (yssource) [reply recipes/dataframe/all/CMakeLists.txt]: Update in the latest commit

**Comment by yssource** [comment]
@danimtb 
Hi, It is OK going to merge this PR, isn't it?

**Pull request merged**